src/MicrosoftTeams.ts

```diff
@@ -1547,20 +1748,25 @@ namespace microsoftTeams {
   function sendMessageRequest(
     targetWindow: Window,
     actionName: string,
-    // tslint:disable-next-line:no-any
+    // tslint:disable-next-line: no-any
     args?: any[]
   ): number {
     let request = createMessageRequest(actionName, args);
-    let targetOrigin = getTargetOrigin(targetWindow);
-
-    // If the target window isn't closed and we already know its origin, send the message right away; otherwise,
-    // queue the message and send it after the origin is established
-    if (targetWindow && targetOrigin) {
-      targetWindow.postMessage(request, targetOrigin);
+    if (isFramelessWindow) {
+      currentWindow.nativeInterface.framelessPostMessage(
+        JSON.stringify(request)
```

Comment:
Are your sure that you don’t need a queue here? According to @mevinay the injection of the framelessPostMessage API in Android would be asynchronous and might not be in place when the SDK starts making calls.

Reply:
@ydogandjiev we found a way on Android which seems to work till now. If at all it fails on any devices, we will have to go back to the earlier approach.

Reply:
Yes, the queue is not needed anymore. We changed approach on the client side. Instead of injecting JS, we have exposed JavaInterface for JS.